pkg/cfn/manager/delete_tasks.go

```diff
@@ -178,26 +181,60 @@ func (c *StackCollection) NewTaskToDeleteUnownedNodeGroup(ctx context.Context, c
 }
 
 // NewTasksToDeleteOIDCProviderWithIAMServiceAccounts defines tasks required to delete all of the iamserviceaccounts
-// along with associated IAM ODIC provider
-func (c *StackCollection) NewTasksToDeleteOIDCProviderWithIAMServiceAccounts(ctx context.Context, oidc *iamoidc.OpenIDConnectManager, clientSetGetter kubernetes.ClientSetGetter) (*tasks.TaskTree, error) {
+// along with associated IAM OIDC provider
+func (c *StackCollection) NewTasksToDeleteOIDCProviderWithIAMServiceAccounts(ctx context.Context, newOIDCManager NewOIDCManager, cluster *ekstypes.Cluster, clientSetGetter kubernetes.ClientSetGetter, force bool) (*tasks.TaskTree, error) {
 	taskTree := &tasks.TaskTree{Parallel: false}
 
-	allServiceAccountsWithStacks, err := c.getAllServiceAccounts(ctx)
+	oidc, err := newOIDCManager()
 	if err != nil {
-		return nil, err
+		if _, ok := err.(*iamoidc.UnsupportedOIDCError); ok {
+			logger.Debug("OIDC is not supported for this cluster")
+			return taskTree, nil
+		}
+		return nil, fmt.Errorf("error creating OIDC manager: %w", err)
 	}
-	saTasks, err := c.NewTasksToDeleteIAMServiceAccounts(ctx, allServiceAccountsWithStacks, clientSetGetter, true)
+
+	allServiceAccountsWithStacks, err := c.getAllServiceAccounts(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	if saTasks.Len() > 0 {
-		saTasks.IsSubTask = true
-		taskTree.Append(saTasks)
+	if len(allServiceAccountsWithStacks) > 0 {
+		saTasks, err := c.NewTasksToDeleteIAMServiceAccounts(ctx, allServiceAccountsWithStacks, clientSetGetter, true)
+		if err != nil {
+			return nil, err
+		}
+
+		if saTasks.Len() > 0 {
+			saTasks.IsSubTask = true
+			taskTree.Append(saTasks)
+		}
 	}
 
 	providerExists, err := oidc.CheckProviderExists(ctx)
 	if err != nil {
+		if iamoidc.IsAccessDeniedError(err) {
+			clusterHasOIDC, foundTag := clusterHasOIDCProvider(cluster)
+			errMsg := "IAM permissions are required to delete OIDC provider"
+			switch {
+			case foundTag:
+				if clusterHasOIDC {
+					return nil, fmt.Errorf("%s: %w", errMsg, err)
+				}
+				if len(allServiceAccountsWithStacks) > 0 {
+					logger.Warning("expected an OIDC provider to be associated with the cluster; found %d service account(s)", len(allServiceAccountsWithStacks))
+				}
+			case len(allServiceAccountsWithStacks) > 0:
+				if !force {
+					return nil, fmt.Errorf("found %d IAM service account(s); %s: %w", len(allServiceAccountsWithStacks), errMsg, err)
+				}
+			default:
+				logger.Info("could not determine if cluster has an OIDC provider because IAM permissions are missing; " +
+					"if an OIDC provider was associated with the cluster (either by setting `iam.withOIDC: true` or by using `eksctl utils associate-iam-oidc-provider`), " +
+					"run `aws iam delete-open-id-connect-provider` from an authorized IAM entity")
```

Comment:
```suggestion
					"run `aws iam delete-open-id-connect-provider` from an authorised IAM entity to clean up remaining resources")
```

Reply:
We have more usages of the US variant of the spelling. Good suggestion on adding the cleanup part, although I'm thinking of:

```suggestion
					"run `aws iam delete-open-id-connect-provider` from an authorized IAM entity to delete the OIDC provider")
```

Reply:
I have changed it to:

```suggestion
					"run `aws iam delete-open-id-connect-provider` from an authorized IAM entity to delete it")
```


so now the message is "if an OIDC provider was associated with the cluster (either by setting `iam.withOIDC: true` or by using `eksctl utils associate-iam-oidc-provider`), run `aws iam delete-open-id-connect-provider` from an authorized IAM entity to delete it".